src/hexen/p_map.c

```
@@ -1529,6 +1547,14 @@ void P_BounceWall(mobj_t * mo)
     P_PathTraverse(leadx, leady, leadx + mo->momx, leady + mo->momy,
                    PT_ADDLINES, PTR_BounceTraverse);
 
+    // P_BounceWall call on a tall sector after fresh game start
+    // without the player sliding along any walls before.
+    // For more details check:
+    // https://github.com/chocolate-doom/chocolate-doom/issues/1732
+    // https://github.com/chocolate-doom/chocolate-doom/issues/1160
+    if (bestslideline == NULL)
+        I_Error("P_BounceWall: No bestslideline was set. Try bumping walls.");
```

Is this correct since vanilla doesn't crash (usually) in this situation? Are there any demos which depend on not crashing here?

---

Yes, such a demo can be created, although it requires either an Heresiarch on level start or the player tiptoing through the map getting hold of the first throwing bomb as a fighter without doing a wallslide. With this Error, I didn't want to fix the issue in the case of demo-recording / new game, but give the player an actual error message instead of a segfault in `P_PointOnLineSide`.

Since we established in the ticket that this can not really be emulated correctly, my decision was to let it crash instead of having the (probably very small) risk of a desync. 😉 That is why I only applied the fix when starting from a savegame, as for those scenarios, any bestslidelines of previous sessions / demos are used anyways and thus is not demo relevant in my opinion.